src/state/apis/zapper/zapperApi.ts

```diff
@@ -325,17 +323,13 @@ export const zapper = createApi({
           }
 
         try {
-          const accountIds = selectWalletAccountIds(state)
-
-          if (!accountIds.length) throw new Error('Not ready')
+          if (!evmAccountIds.length) throw new Error('Not ready')
```

Comment:
It seems odd this query should know an empty list of account ids would mean upstream is not ready?

Reply:
I mean we could just as well return empty results - the result will effectively be the same since a new query will be re-ran on args change. That's assuming `refetchOnMountOrArgChange: true` though which wasn't true previously.

Would you find returning empty results to be better semantics?

Reply:
Yes I think so. It makes sense that requesting an empty set of IDs would do one of:
- return empty results
- or if you want to implement error handling, return a non-context specific error like `evmAccountIds should not be empty` and let the consumer decide what that means to the implementation

Reply:
Actually agree here, semantically that's not an error.
The only valid case for returning an error in RTK queries when not ready is consumooors not refetching on args change (which is odd to have as a default in RTK, you'd expect the default behavior is refetching on args change but nah) - which isn't the case anymore, our only consumor is now refetching on args change so no reason to return an RTK error anymore!


07f3874d05 and its Jam retest https://jam.dev/c/887eca6a-42c9-44ce-b9c3-784f965a9c86

Note I kept the default case in `selectGetReadOnlyOpportunities`, as we may not have any success query, or may have only legit failures.